### DISCUSSION TITLE
Script version 3.3.0 doesn't add any signatures to Outlook *Fixed*

### DISCUSSION THREAD

**Discussion by KoenBouten** [Set-OutlookSignatures/Set-OutlookSignatures#73] [General]
With version 3.3.0 of the script I encountered an error while running the script (see screenshot).
This error prevents the script from adding the signatures to Outlook.
I fixed this by removing the empty brackets at line 1022 of the script:

Before:
$MailAddresses = @($ADPropsCurrentUser.mail.tolower()) + $MailAddresses

After:
$MailAddresses = @($ADPropsCurrentUser.mail.tolower) + $MailAddresses

It took me a while to figure out since I'm new to all of this, so I thought I would write this discussion just in case that someone else would struggle with this problem.

![2023-04-21 14_27_03-New Discussion · GruberMarkus_Set-OutlookSignatures](https://user-images.githubusercontent.com/117994540/233635920-55c2246b-c4bc-4d25-979f-d98842b8902b.png)

**Comment by GruberMarkus** [comment]
Version 3.3.0 is old. How does the current version 3.6.0 behave?

**Reply by KoenBouten** [reply]
Before I started using version 3.3.0 I was using 3.6.0, but when I used that version the script wouldn't add the signatures I made to the shared mailboxes I have. But then my supervisor said that it worked just fine when he executed the script. Later we realised he was using version 3.3.0, and I was using version 3.6.0. So then I also started using version 3.3.0 and that fixed my problem. We both still don't know what the problem was (because I'm new to powershell, html, etc.). So that is the reason why I'm using version 3.3.0 (for now). So now I only get these kind of error's which I can fix (relatively easy) myself.

**Comment by GruberMarkus** [comment]
Please also be aware that your code from post #1 does not fix any problem, but avoid one and create others.  The difference between `$ADPropsCurrentUser.mail.tolower()` and `$ADPropsCurrentUser.mail.tolower` is that the first command converts the mail attribute string to lower case letters, while the second one tries to access the attribute mail.tolower, which does not exist and is null in any case.

As the logic behind this code has changed in more recent versions, please send me the whole output of v3.6.0 showing the error with shared mailboxes. Make sure that you use the "-verbose" parameter.

**Reply by KoenBouten** [reply]
But that is also a bit of an issue, because it doesn't give an error at the 'Get e-mail addresses from Outlook profiles and corresponding registry paths' part. It only displays my primary mailbox. While at version 3.3.0, it also finds/sees the shared mailboxes.
So I am a bit confused on why it does work on version 3.3.0, but not on version 3.6.0. (I do still have some errors when I execute the script, but they are the same on both versions, and they are from different parts of the script. So I don't know why one works and the other doesn't)

**Reply by GruberMarkus** [reply]
The root cause can not be found without a verbose log.

**Reply by KoenBouten** [reply]
This is what it says:
![image](https://user-images.githubusercontent.com/117994540/234253301-0561782b-c22c-4df8-9460-5c517e89219f.png)

But I don't know where and how I can fix it.
(I haven't edited anything in the 'Set-Outlooksignatures' script)

**Reply by GruberMarkus** [reply]
The message posted contains a clear hint of the root cause: The mail attribute of the currently logged on user is not set in Active Directory, and no mailbox in trusted domains has been found via msExchMasterAccountSid.

There are several posiible root causes for this, but I can't say much more if you do not provide the requested **whole** output of v3.6.0 showing the error with shared mailboxes. Make sure that you use the "-verbose" parameter. Please do not transmit the log as screenshot, but as text file.

The sort order message is an internal bug that has already been corrected in the code of the upcoming release v4.0.0. In your case, mailboxes are not processed according to the display sort order in Outlook, but according to the timely sequence in which they have been added to Outlook (as documented in the README file).

**Reply by KoenBouten** [reply]
I don't know if I can send you the whole ouptut as a text file, because it may contain some information from our customers that I am not allowed to share. So I would have to ask my supervisor about that.
But until then I am only able to show you through screenshots unfortunately.

**Reply by GruberMarkus** [reply]
If you can't sanitize the logs or need implementation support covering the documentation or your specific environment, there is an option for paid commercial remote support.

I will close this discussion for now. Please create a new issue for this problem when you can share the full verbose logs of the most recent version.
